### DISCUSSION TITLE
Turning off error middle ware dynamically

### DISCUSSION THREAD

**Discussion by fhanggi** [reduxjs/redux-toolkit#4514] [Q&A]
Hello

I have the  error middle ware below which handles all errors for any api call made. We have a scenario now where we would like the error to be handled in the code and not in the middleware.
How could this be done?

i thought i could maybe set some meta flag in the component that wants to handle the error but did not succeed.

Like i though i could change the if statement in the middleware to this and then set that meta somewhere
    if (isRejectedWithValue(action) && !action.meta?.handleErrorDirectly ) { 

but i did not succeed setting that meta flag.

Any advice would be greatly appreciated. Thank you

```
import { isRejectedWithValue } from "@reduxjs/toolkit";
import history from "../history";
import { createApplicationNotification } from "../redux/AppNotificationSlice";

export function rtkQueryErrorHandler(api) {
  return (next) => (action) => {
    if (isRejectedWithValue(action)) {
      console.log("We got a rejected action!", action);
      const status = action.payload?.status;
      const errorMessage = action.error?.message;
      const url = action.meta?.baseQueryMeta?.request?.url;
      if (status && ![481, 482].includes(status)) {
        console.log("Error status:", status, "Action:", action);
        console.log(action.payload);
        const errorInfo = {
          message: `Request failed with status ${status} | error: ${errorMessage} | url: ${url}`,
          status: status,
          // payload: action.payload,
          // genericMessage: "An error occurred in RTKQ Middleware.",
          showHeader: false
        };

        //navigate to the global error page
        history.push("/error", { error: errorInfo });
        return next(action);
      }

      if (status && [481, 482].includes(status)) {
        console.log(action.payload);
        const errorInfo = action.payload.data;
        console.log(errorInfo);
        if (status === 481 && errorInfo.detail) {
          const errorDetails = errorInfo.detail.split("|");
          const errorTypes = errorInfo.type.split("|");
          errorDetails.forEach((errorDetail, index) => {
            api.dispatch(createApplicationNotification(errorDetail, errorTypes[index]));
          });
        }
        return next(action);
      }
    }

    return next(action);
  };
}

```

**Comment by fhanggi** [comment]
I have found one way to skip the error middleware but it seems a bit cluncky

pass in an extra parameter to the query (skipGlobalErrorHandling) which will be ignored by the query itself but seems available in the originalArgs object in the error middleware

    triggerFetchApplication({ id: "777777", skipGlobalErrorHandling: true })
      .unwrap()
      .catch((e) => {
        console.log(e);
        dispatch(createApplicationNotification(e.data.title, "error"));
      });

in error middleware only do the error handling if below is true
    if (isRejectedWithValue(action) && !action.meta?.arg?.originalArgs?.skipGlobalErrorHandling) {
    
Is this an ok thing to do ? 
Is there something better?

Thank you

**Reply by fhanggi** [reply]
Anyone have any suggestions?
Thank you very much.

**Comment by lamhungypl** [comment]
From my experiences https://github.com/reduxjs/redux-toolkit/discussions/4301, you'll need to set meta info in the query definition
